<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>903</width>
    <height>731</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSplitter" name="log_box_splitter">
      <property name="orientation">
       <enum>Qt::Orientation::Vertical</enum>
      </property>
      <widget class="QWidget" name="main_widget" native="true">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QSplitter" name="splitter">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <widget class="QWidget" name="HV_LED_widget" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <property name="spacing">
             <number>0</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QGroupBox" name="LED_groupBox">
              <property name="title">
               <string>LED Control</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_8">
               <item row="0" column="1">
                <widget class="QPushButton" name="LED_power_pushButton">
                 <property name="text">
                  <string>Power LEDs ON</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="LED_state_label">
                 <property name="styleSheet">
                  <string notr="true">font: 900 9pt &quot;Segoe UI&quot;;</string>
                 </property>
                 <property name="text">
                  <string>OFF</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignmentFlag::AlignCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="HV_groupBox">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>High Voltage Control</string>
              </property>
              <layout class="QGridLayout" name="gridLayout">
               <item row="4" column="0" colspan="2">
                <spacer name="HV_verticalSpacer">
                 <property name="orientation">
                  <enum>Qt::Orientation::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item row="3" column="0" colspan="2">
                <widget class="QGroupBox" name="HV_connected_groupBox">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_2">
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                  <property name="spacing">
                   <number>6</number>
                  </property>
                  <item row="1" column="1">
                   <widget class="QLabel" name="HV_target_voltage_label">
                    <property name="text">
                     <string>Target Voltage [V]:</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="2">
                   <widget class="QSpinBox" name="HV_target_voltage_spinBox">
                    <property name="maximum">
                     <number>30000</number>
                    </property>
                    <property name="singleStep">
                     <number>1000</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="3">
                   <widget class="QPushButton" name="HV_set_target_voltage_pushButton">
                    <property name="text">
                     <string>Set</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="2" colspan="2">
                   <widget class="QPushButton" name="HV_enable_pushButton">
                    <property name="text">
                     <string>Enable</string>
                    </property>
                    <property name="checkable">
                     <bool>true</bool>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1">
                   <widget class="QLabel" name="HV_state_label">
                    <property name="styleSheet">
                     <string notr="true">font: 900 9pt &quot;Segoe UI&quot;;</string>
                    </property>
                    <property name="text">
                     <string>OFF</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1" colspan="3">
                   <widget class="QLabel" name="HV_live_voltage_label">
                    <property name="styleSheet">
                     <string notr="true">font: 24pt &quot;Segoe UI&quot;;</string>
                    </property>
                    <property name="text">
                     <string>Voltage: NaN V</string>
                    </property>
                    <property name="textFormat">
                     <enum>Qt::TextFormat::PlainText</enum>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignCenter</set>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QCheckBox" name="HV_power_checkBox">
                 <property name="text">
                  <string>Power</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QPushButton" name="HV_connect_pushButton">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Connect</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QGroupBox" name="positioning_groupBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Positioning Control</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QCheckBox" name="positioning_power_checkBox">
              <property name="text">
               <string>Power</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="2">
             <widget class="QWidget" name="positioning_homing_done_widget" native="true">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QWidget" name="positioning_experiment_running_widget" native="true">
                 <layout class="QVBoxLayout" name="verticalLayout_4">
                  <property name="spacing">
                   <number>0</number>
                  </property>
                  <property name="leftMargin">
                   <number>0</number>
                  </property>
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <property name="rightMargin">
                   <number>0</number>
                  </property>
                  <property name="bottomMargin">
                   <number>0</number>
                  </property>
                  <item>
                   <widget class="QGroupBox" name="positioning_pump_1_groupBox">
                    <property name="title">
                     <string>Pump 1</string>
                    </property>
                    <layout class="QGridLayout" name="gridLayout_4">
                     <item row="1" column="3">
                      <widget class="QPushButton" name="positioning_pump_1_move_forward_10_pushButton">
                       <property name="text">
                        <string>+10 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="1">
                      <widget class="QPushButton" name="positioning_pump_1_move_back_1_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -1</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="0" colspan="2">
                      <widget class="QLabel" name="positioning_pump_1_flow_label">
                       <property name="text">
                        <string>Flow [ml/h]:</string>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="0">
                      <widget class="QPushButton" name="positioning_pump_1_move_back_10_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -10</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="2" colspan="2">
                      <widget class="QDoubleSpinBox" name="positioning_pump_1_flow_doubleSpinBox">
                       <property name="decimals">
                        <number>1</number>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="2">
                      <widget class="QPushButton" name="positioning_pump_1_move_forward_1_pushButton">
                       <property name="text">
                        <string>+1 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </widget>
                  </item>
                  <item>
                   <widget class="QGroupBox" name="positioning_pump_2_groupBox">
                    <property name="title">
                     <string>Pump 2</string>
                    </property>
                    <layout class="QGridLayout" name="gridLayout_5">
                     <item row="1" column="3">
                      <widget class="QPushButton" name="positioning_pump_2_move_forward_10_pushButton">
                       <property name="text">
                        <string>+10 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="0">
                      <widget class="QPushButton" name="positioning_pump_2_move_back_10_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -10</string>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="2">
                      <widget class="QPushButton" name="positioning_pump_2_move_forward_1_pushButton">
                       <property name="text">
                        <string>+1 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="0" colspan="2">
                      <widget class="QLabel" name="positioning_pump_2_flow_label">
                       <property name="text">
                        <string>Flow [ml/h]:</string>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="2" colspan="2">
                      <widget class="QDoubleSpinBox" name="positioning_pump_2_flow_doubleSpinBox">
                       <property name="decimals">
                        <number>1</number>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="1">
                      <widget class="QPushButton" name="positioning_pump_2_move_back_1_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -1</string>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </widget>
                  </item>
                  <item>
                   <widget class="QGroupBox" name="positioning_stage_groupBox">
                    <property name="title">
                     <string>Stage</string>
                    </property>
                    <layout class="QGridLayout" name="gridLayout_6">
                     <item row="3" column="3" colspan="2">
                      <widget class="QPushButton" name="positioning_stage_calibrate_center_pushButton">
                       <property name="toolTip">
                        <string>Move the stage to the left-most position and press &quot;Calibrate Center&quot;.
 New center is in the middle of current position and homing position.</string>
                       </property>
                       <property name="text">
                        <string>Calibrate Center</string>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="3" colspan="2">
                      <widget class="QSpinBox" name="positioning_stage_amplitude_spinBox">
                       <property name="maximum">
                        <number>100</number>
                       </property>
                       <property name="singleStep">
                        <number>10</number>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="1">
                      <widget class="QPushButton" name="positioning_stage_move_back_10_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -10</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="1" colspan="2">
                      <widget class="QLabel" name="positioning_stage_speed_label">
                       <property name="text">
                        <string>Speed [mm/min]:</string>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="3">
                      <widget class="QPushButton" name="positioning_stage_move_forward_1_pushButton">
                       <property name="text">
                        <string>+1 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="4">
                      <widget class="QPushButton" name="positioning_stage_move_forward_10_pushButton">
                       <property name="text">
                        <string>+10 &gt;&gt;</string>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="2">
                      <widget class="QPushButton" name="positioning_stage_move_back_1_pushButton">
                       <property name="text">
                        <string>&lt;&lt; -1</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="3" colspan="2">
                      <widget class="QSpinBox" name="positioning_stage_speed_spinBox">
                       <property name="maximum">
                        <number>2000</number>
                       </property>
                       <property name="singleStep">
                        <number>100</number>
                       </property>
                       <property name="value">
                        <number>1000</number>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="1" colspan="2">
                      <widget class="QLabel" name="positioning_stage_amplitude_label">
                       <property name="text">
                        <string>Amplitude [mm]:</string>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                       </property>
                      </widget>
                     </item>
                     <item row="3" column="1" colspan="2">
                      <widget class="QPushButton" name="positioning_stage_move_to_center_pushButton">
                       <property name="text">
                        <string>Move to Center</string>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <widget class="QGroupBox" name="positioning_experiment_groupBox">
                 <property name="title">
                  <string>Experiment</string>
                 </property>
                 <layout class="QGridLayout" name="gridLayout_7">
                  <item row="0" column="0">
                   <widget class="QLabel" name="positioning_experiment_duration_label">
                    <property name="text">
                     <string>Duration [min]:</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="0">
                   <widget class="QPushButton" name="positioning_experiment_start_pushButton">
                    <property name="text">
                     <string>Start</string>
                    </property>
                   </widget>
                  </item>
                  <item row="2" column="1">
                   <widget class="QPushButton" name="positioning_experiment_stop_pushButton">
                    <property name="text">
                     <string>Stop</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QSpinBox" name="positioning_experiment_duration_spinBox">
                    <property name="toolTip">
                     <string>For infinite experiment duration set &quot;0&quot;</string>
                    </property>
                    <property name="maximum">
                     <number>600</number>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="positioning_experiment_remaining_time_label">
                    <property name="text">
                     <string>Remaining Time [min:sec]:</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="positioning_experiment_remaining_time_value_label">
                    <property name="text">
                     <string>NaN</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="positioning_home_pushButton">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Home</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <spacer name="positioning_verticalSpacer">
              <property name="orientation">
               <enum>Qt::Orientation::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="2" column="0" colspan="2">
             <widget class="QGroupBox" name="positioning_send_command_groupBox">
              <property name="title">
               <string>DEV - Send Command</string>
              </property>
              <layout class="QGridLayout" name="gridLayout_9">
               <item row="0" column="0">
                <widget class="QLineEdit" name="positioning_send_command_lineEdit"/>
               </item>
               <item row="0" column="1">
                <widget class="QPushButton" name="positioning_send_command_pushButton">
                 <property name="text">
                  <string>Send</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="log_box_groupBox">
       <property name="title">
        <string>LogBox</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>11</number>
        </property>
        <property name="topMargin">
         <number>11</number>
        </property>
        <property name="rightMargin">
         <number>11</number>
        </property>
        <property name="bottomMargin">
         <number>11</number>
        </property>
        <item>
         <widget class="QTextBrowser" name="log_box_textBrowser"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>903</width>
     <height>33</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
